addons/pos_loyalty/static/src/js/Loyalty.js

```diff
@@ -185,18 +190,62 @@ const PosLoyaltyOrderline = (Orderline) => class PosLoyaltyOrderline extends Ord
                 }
             });
             for (const line of linesToRemove) {
-                this.order.orderlines.remove(line);
+                this.order.remove_orderline(line);
             }
         }
         return super.set_quantity(...arguments);
     }
+    /**
+     * @override
+     */
+    can_be_merged_with(otherLine) {
+        if (this.is_reward_line && otherLine.is_reward_line) {
+            if (otherLine.reward_id == this.reward_id) {
+                const reward = this.order.pos.reward_by_id[this.reward_id];
+                if (reward && reward.reward_type == 'product') {
+                    return otherLine.product.id == this.product.id;
+                }
+            }
+        }
+        return this.is_reward_line == otherLine.is_reward_line && super.can_be_merged_with(otherLine);
+    }
+    /**
+     * Returns the points consumed by this reward line.
+     * - If there is `force_points_cost`, we return that.
+     * - Otherwise, we compute based on the reward object
+     *   and quantity of this orderline.
+     */
+    get points_cost() {
```

Comment:
don't use those kind of js bs getter, it's hella confusing because I (and other) will think that `this.points_cost` is an attribute that can be overwritten. Make it as a normal function